src/core/libmaven/groupFiltering.cpp

```diff
@@ -124,6 +122,76 @@ bool GroupFiltering::filterByMS2(PeakGroup& peakgroup)
     return false;
 }
 
+void GroupFiltering::filterBasedOnParent(PeakGroup& parent,
+                                         GroupFiltering::ChildFilterType type,
+                                         float maxRtDeviation,
+                                         float minPercentCorrelation,
+                                         MassCutoff* massCutoff)
+{
+    auto possibleChildren = parent.childIsotopes();
+    if (type == ChildFilterType::BarplotIsotope) {
+        possibleChildren = parent.childIsotopesBarPlot();
+    } else if (type == ChildFilterType::Adduct) {
+        possibleChildren = parent.childAdducts();
+    }
+
+    vector<PeakGroup*> nonChildren;
+    for (auto& child : possibleChildren) {
+        float rtDeviationSum = 0.0f;
+        int numSamplesShared = 0;
+        for (auto sample : _mavenParameters->samples) {
+            auto childPeak = child->getPeak(sample);
+            auto parentPeak = parent.getPeak(sample);
+            if (parentPeak == nullptr || childPeak == nullptr)
+                continue;
+
+            ++numSamplesShared;
+            auto childApexRt = childPeak->rt;
+            auto parentApexRt = parentPeak->rt;
+            auto deviation = abs(childApexRt - parentApexRt) * 60.0f;
```

Comment:
Could you add a comment here to explain that the multiplication with 60 is being done to convert the time unit? it's pretty obvious but still.

Reply:
Okay.